src/uci.cpp

```diff
@@ -568,7 +568,7 @@ bool UCI::is_valid_option(UCI::OptionsMap& options, std::string& name) {
   std::string protocol = options["Protocol"];
   for (const auto& it : options)
   {
-      if (options.key_comp()(option_name(it.first, protocol), name))
+      if (option_name(it.first, protocol) == name)
```

Comment:
Since we are comparing strings here and not an Option to a string, we actually have to use the case insensitive less comparator instead of the equality operator in order to avoid having a case sensitive match. I think my version was wrong because the comparator is testing for "less", not for equality. So something like the below should work I think.
```suggestion
      std::string optionName = option_name(it.first, protocol);
      if (!options.key_comp()(optionName, name) && !options.key_comp()(name, optionName))
```